client/settings/payment-settings/test-mode-checkbox.js

```diff
@@ -0,0 +1,143 @@
+import { __ } from '@wordpress/i18n';
+import { React, useState } from 'react';
+import { Button, CheckboxControl } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+import { useTestMode } from 'wcstripe/data';
+import { useAccountKeys } from 'wcstripe/data/account-keys';
+import ConfirmationModal from 'wcstripe/components/confirmation-modal';
+import InlineNotice from 'wcstripe/components/inline-notice';
+
+const MissingAccountKeysModal = ( { type, onClose } ) => {
+	const [ , setTestMode ] = useTestMode();
+
+	const handleSave = () => {
+		setTestMode( type === 'test' );
+		onClose();
+	};
+
+	return (
+		<ConfirmationModal
+			onRequestClose={ onClose }
+			actions={
+				<>
+					<Button isSecondary onClick={ onClose }>
+						{ __( 'Cancel', 'woocommerce-gateway-stripe' ) }
+					</Button>
+					<Button isPrimary onClick={ handleSave }>
+						{ __( 'Save changes', 'woocommerce-gateway-stripe' ) }
+					</Button>
+				</>
+			}
+			title={
+				type === 'test'
+					? __(
+							'Edit test account keys & webhooks',
+							'woocommerce-gateway-stripe'
+					  )
+					: __(
+							'Edit live account keys & webhooks',
+							'woocommerce-gateway-stripe'
+					  )
+			}
+		>
+			<InlineNotice isDismissable={ false }>
+				{ type === 'test'
+					? interpolateComponents( {
+							mixedString: __(
+								"To enable the test mode, get the test account keys from your {{accountLink}}Stripe Account{{/accountLink}} (we'll save them for you so you won't have to do this every time).",
+								'woocommerce-gateway-stripe'
+							),
+							components: {
+								accountLink: (
+									// eslint-disable-next-line jsx-a11y/anchor-has-content
+									<a href="https://dashboard.stripe.com/test/apikeys" />
+								),
+							},
+					  } )
+					: interpolateComponents( {
+							mixedString: __(
+								"To enable the live mode, get the account keys from your {{accountLink}}Stripe Account{{/accountLink}} (we'll save them for you so you won't have to do this every time).",
+								'woocommerce-gateway-stripe'
+							),
+							components: {
+								accountLink: (
+									// eslint-disable-next-line jsx-a11y/anchor-has-content
+									<a href="https://dashboard.stripe.com/apikeys" />
+								),
+							},
+					  } ) }
+			</InlineNotice>
+		</ConfirmationModal>
+	);
+};
+
+const TestModeCheckbox = () => {
+	const [ modalType, setModalType ] = useState( '' );
+	const [ isTestModeEnabled, setTestMode ] = useTestMode();
+	const { accountKeys } = useAccountKeys();
+
+	const handleCheckboxChange = ( isChecked ) => {
+		// are we enabling test mode without the necessary keys?
+		if (
+			isChecked &&
+			( ! accountKeys.test_publishable_key ||
+				! accountKeys.test_secret_key ||
+				! accountKeys.test_webhook_secret )
```

Comment:
QQ: is the `webhook_secret` a necessary field like `secret_key` and `publishable_key`?

Reply:
Yeah, it's something merchants should enter, otherwise their order status might not be working correctly.
In the new onboarding flow we're going to require it.